Navbar

to-sta opened the pull request:
<!---
Thank you for your pull request! 🚀
-->

### Contributor checklist

<!-- Please replace the empty checkboxes [] below with checked ones [x] accordingly. -->

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

This PR fixes a `Navbar` issue on small screens. On small screens if the `SearchBar` is expanded it could overlap the burger menu on the right side. Now the expanded `Searchbar` takes the remaining space next to burger menu.

This PR fixes the issues that arises on small screens if the searchbar is expanded, it can overlap the burger menu. The searchbar had a fixed width, now the searchbar is responsive and takes the remaining space next to the burger menu. The space between the burger menu and the expanded searchbar can be adjusted with the `gap-`

I have check the landing page and /home with different browsers:

- Mozilla Firefox
- Chrome 
- Edge

### Related issue

#249

andrewtavis commented on frontend/components/SearchBar.vue:
```diff
@@ -5,30 +5,18 @@
-        class="absolute right-0 flex pr-6 space-x-1 transition-opacity transition-duration-200"
-      >
+      <div v-if="sidebar.collapsed == false || sidebar.collapsedSwitch == false" ref="hotkeyIndicators"
+        class="absolute right-0 flex pr-2 space-x-1 transition-opacity transition-duration-200">
```
I think that this should be maintained at `pr-6`, but it's also mainly because of a change I sent through in the interim. I remember that we were talking about making this relative to the contianer, but then it makes it so that the tooltips get cut off at the edge of where the searchbar's margins intersect with the sidebar header.

If you have another suggestion I'd be happy to consider it!
```suggestion
        class="absolute right-0 flex pr-2 space-x-1 transition-opacity transition-duration-200">
```

andrewtavis commented on frontend/components/SearchBar.vue:
```diff
@@ -5,30 +5,18 @@
-        class="absolute right-0 flex pr-6 space-x-1 transition-opacity transition-duration-200"
-      >
+      <div v-if="sidebar.collapsed == false || sidebar.collapsedSwitch == false" ref="hotkeyIndicators"
+        class="absolute right-0 flex pr-2 space-x-1 transition-opacity transition-duration-200">
```
I think that this should be maintained at pr-6, but it's also mainly because of a change I sent through in the interim. I remember that we were talking about making this relative to the contianer, but then it makes it so that the tooltips get cut off at the edge of where the searchbar's margins intersect with the sidebar header.

If you have another suggestion I'd be happy to consider it!

andrewtavis replied:
This is what the tooltip looks like for me locally with `relative` in the top div that I removed from here and pr-2:

![Screenshot_2023-08-12 01 06 54](https://github.com/activist-org/activist/assets/24387426/df215fe3-ccd0-4015-8a32-059ca1af4098)

andrewtavis commented on frontend/components/WebsiteHeader.vue:
```diff
@@ -1,21 +1,14 @@
-            :is-mobile="true"
-            class="absolute inset-0 flex items-center justify-center z-1 overflow-clip"
-          />
+    <div class="flex md:hidden py-3 px-4">
```
Another minor thing I'm wondering about is why for the website header on mobile the focus for it is now the entire container rather than just `activist` 🤔

![Screenshot_2023-08-12 01 12 58](https://github.com/activist-org/activist/assets/24387426/69171a20-9350-4248-a23e-60f3248c627c)

Seems to be coming from here. Generally any edit that gets the focus back properly would be great 😊

andrewtavis commented:
These were the only major things that jumped out to me, @to-sta 😊 Thanks for the work on this! 🙌

to-sta commented:
@andrewtavis thanks for the review. I will revise my changes to include your findings.

andrewtavis commented:
Thanks, @to-sta! If you can find a way to maintain `relative` for the search bar then that would likely be best. Maybe playing around with `z-` a bit could bring the tooltip out :)

to-sta commented:
@andrewtavis fixed the focus on the `LogoActivist`. Figured out that the `NuxtLink` creates the anchar tag and the anchor tag inherits the size of the parent element. Added a div that solves that issues. 

Added the pr-6 back in again for the `SearchBar`. I could not reproduce the issue with the tooltip. Try it in firefox, chrome and edge. Always saw the tooltips. I am not sure how i can reproduce the issue you encountered.

andrewtavis commented:
@to-sta, sorry, I didn't do a good job of explaining a change I made in 1b03d73 that's preventing you from reproducing the issue. You'd need to add in `relative` to the top `<div>` for the search bar. This `relative` was the Tailwind class that was allowing us to place the `/` in relation to the search bar rather than in relation to the sidebar itself. The `pr-6` is applied as `pr-0` puts us at the sidebar's edge, whereas with `relative` `pr-0` goes to the edge of the search bar. With relative I was also getting this error with the `/` tooltip. I think that this works for now, and my suggestion of adding in a `z-INT` class won't actually work as that's already being applied by the following custom class:

https://github.com/activist-org/activist/blob/6086585fcac56b9a581acc78d8bcac84305c8d7e/frontend/assets/css/tailwind.css#L38

Thanks for the changes! Both things are working great 😊 Let me fix the merge conflicts and we'll be good to go 🚀

andrewtavis reviewed (approved):
Thanks for the quick changes here, @to-sta! Again I really appreciate the diagnosis and fixing this. Will be a lot happier looking at activist on my phone now 😊
